assemblyline/filestore/transport/s3.py

```diff
@@ -65,12 +65,14 @@ def __init__(self, base=None, accesskey=None, secretkey=None, aws_region=None, s
         )
 
         bucket_exist = False
-        resp = self.with_retries(self.client.list_buckets)
-
-        for bucket in resp["Buckets"]:
-            if bucket.get("Name", None) == self.bucket:
-                bucket_exist = True
-                break
+        try:
+            self.client.head_bucket(Bucket=self.bucket)
```

Comment:
While this will definitely works with Amazon S3, it may cause some issues when using with minio where as the minio server may not be already available. It would be preferable to wrap the call in the with_retries function to the call is blocked until minio comes online.

Reply:
You have a good exemple how to do it here: https://github.com/CybercentreCanada/assemblyline-base/blob/f7f18c29fda129627740d0499e4b7235f1951552/assemblyline/filestore/transport/s3.py#L126

Reply:
Can do. Let me test a change that incorporates `with_retries` and then I'll push it.

Reply:
Tested and committed using `with_retries`.  The test was simple:
1. Verify normal operation.
2. Attempt to connect to a non-existent S3 instance and observe retries.

```python
>>> from assemblyline.filestore.transport.s3 import TransportS3
>>> s3 = TransportS3(…)
No connection to S3 transport http://localhost:80, retrying...
No connection to S3 transport http://localhost:80, retrying...
No connection to S3 transport http://localhost:80, retrying...
No connection to S3 transport http://localhost:80, retrying...
No connection to S3 transport http://localhost:80, retrying...
```